opentelemetry-extra/src/OpenTelemetry/EventlogStreaming_Internal.hs

```diff
@@ -396,6 +406,14 @@ inventSpanId serial st = (st', sid)
     (SId -> sid, randomGen') = R.nextWord64 randomGen
     st' = st {serial2sid = HM.insert serial sid serial2sid, randomGen = randomGen'}
 
+inventDisplayTid :: ThreadId -> State -> (State, ThreadId)
+inventDisplayTid tid st@(S {thread2displayThread}) =
+      case HM.lookup tid thread2displayThread of
+        Nothing ->
+                  let new_dtid = fromIntegral (HM.size thread2displayThread)
```

Comment:
`HM.size` is surprisingly O(n) so you probably want to memoize it

Reply:
Indeed surprising, I didn't realize that.